.github/workflows/deploy.yml

```diff
@@ -4,11 +4,82 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
+env:
+ DOCKERHUB_USERNAME: 'dfedigital'
+ DOCKERHUB_IMAGE_NAME: 'teaching-vacancies'
 
 jobs:
-  deploy:
-    name: Deploy to Production
+  build_image:
+    name: build & publish docker image
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      name: Checkout Code
+
+    - name: prepare env
+      run: |
+        gpg --quiet --batch --yes --decrypt --passphrase="$DOTENV_PASSPHRASE" --output .env dotenv.gpg
+      env:
+        DOTENV_PASSPHRASE: ${{ secrets.DOTENV_PASSPHRASE }}
+
+    - name: docker build
```

Comment:
Could we use the builtin action?
That's what is used [in get into teaching](https://github.com/DFE-Digital/get-into-teaching-app/blob/master/.github/workflows/build.yml#L24)